staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -119,11 +144,28 @@ func CaseSensitiveJsonIterator() jsoniter.API {
 	return config
 }
 
-// Private copy of jsoniter to try to shield against possible mutations
+// CaseSensitiveStrictJsonIterator returns a jsoniterator API that's configured to be
+// case-sensitive, but also disallow unknown fields when unmarshalling. It is compatible with
+// the encoding/json standard library.
+func CaseSensitiveStrictJsonIterator() jsoniter.API {
+	config := jsoniter.Config{
+		EscapeHTML:             true,
+		SortMapKeys:            true,
+		ValidateJsonRawMessage: true,
+		CaseSensitive:          true,
+		DisallowUnknownFields:  true,
```

Comment:
consider OnlyTaggedField=true?

Reply:
From what I understand, that makes a lot of sense. We might want to add a test that verifies that it will not decode anything into a field not specifically marked with a JSON tag.

Reply:
`OnlyTaggedField` seems like a knob to skip untagged fields when marshaling.
i don't see it affecting unmarshaling.

for marshaling do we really want that?
my initial reaction would be no, as this is just a silent skip.

Reply:
it looks like it's used in `decoderOfStruct` as well... I thought it would keep us from decoding things into untagged fields.

that said, that shouldn't really be part of strict decoding, more part of our API type linting, so I'm ok not doing that here